tikv/closure_exec.go

```diff
@@ -0,0 +1,162 @@
+package tikv
+
+import (
+	"bytes"
+	"math"
+
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/types"
+	"github.com/juju/errors"
+	"github.com/pingcap/tipb/go-tipb"
+	"github.com/pingcap/tidb/util/codec"
+)
+
+// tryBuildClosureExecutor tries to build a closureExecutor for the DAGRequest.
+// currently, only 'count(*)' is supported, but we can support all kinds of requests in the future.
+func (svr *Server) tryBuildClosureExecutor(dagCtx *dagContext, dagReq *tipb.DAGRequest) *closureExecutor {
+	executors := dagReq.Executors
+	if len(executors) != 2 {
+		return nil
+	}
+	var cols []*tipb.ColumnInfo
+	var unique bool
+	switch executors[0].Tp {
+	case tipb.ExecType_TypeTableScan:
+		tblScan := executors[0].TblScan
+		cols = tblScan.Columns
+		unique = true
+	case tipb.ExecType_TypeIndexScan:
+		idxScan := executors[0].IdxScan
+		cols = idxScan.Columns
+		unique = idxScan.GetUnique()
```

Comment:
Should we return nil for other Type in default branch?

Reply:
The first executor is always IndexScan or TableScan.

Reply:
So please add comments for assumption